src/super_gradients/training/models/conversion.py

```diff
@@ -48,6 +56,109 @@ def forward(self, x):
         return self.post_process(self.model(self.pre_process(x)))
 
 
+@resolve_param("pre_process", TransformsFactory())
+@resolve_param("post_process", TransformsFactory())
+def convert_to_coreml(
+    model: torch.nn.Module,
+    out_path: str,
+    input_shape: tuple = None,
+    pre_process: torch.nn.Module = None,
+    post_process: torch.nn.Module = None,
+    prep_model_for_conversion_kwargs=None,
+    export_as_ml_program=False,
+    torch_trace_kwargs=None,
+):
+    """
+        Exports a given SG model to CoreML mlprogram or package.
+
+        :param model: torch.nn.Module, model to export to ONNX.
+        :param out_path: str, destination path for the .onnx file.
+        :param input_shape: Input shape without batch dimensions ([C,H,W]). Batch size assumed to be 1.
+        DEPRECATED USE input_size KWARG IN prep_model_for_conversion_kwargs INSTEAD.
+        :param pre_process: torch.nn.Module, preprocessing pipeline, will be resolved by TransformsFactory()
+        :param post_process: torch.nn.Module, postprocessing pipeline, will be resolved by TransformsFactory()
+        :param prep_model_for_conversion_kwargs: dict, for SgModules- args to be passed to model.prep_model_for_conversion
+         prior to torch.onnx.export call. Supported keys are:
+        - input_size - Shape of inputs with batch dimension, [C,H,W] for image inputs.
+         When true, the simplified model will be saved in out_path (default=True).
+        :param export_as_ml_program: Whether to convert to the new program format (better) or legacy coreml proto file
+                            (Supports more iOS versions and devices, but this format will be deprecated at some point).
+        :param torch_trace_kwargs: kwargs for torch.jit.trace
+    :return: Path
+    """
+    if ct is None:
+        raise ImportError(
+            '"coremltools" is required for CoreML export, but is not installed. Please install CoreML Tools using:\n'
+            '   "python3 -m pip install coremltools" and try again (Tested with version 6.3.0);'
+        )
+
+    logger.debug("Building model...")
+    logger.debug(model)
+    logger.debug("Model child nodes:")
+    logger.debug(next(model.named_children()))
+
+    if not os.path.isdir(pathlib.Path(out_path).parent.resolve()):
+        raise FileNotFoundError(f"Could not find destination directory {out_path} for the ONNX file.")
+    torch_trace_kwargs = torch_trace_kwargs or dict()
+    prep_model_for_conversion_kwargs = prep_model_for_conversion_kwargs or dict()
+
+    if input_shape is not None:
```

Comment:
Since this is a new function, I don't think adding a deprecated parameter to it is a right idea. To me it make sense to drop it. I realize this is to mimic the API of `convert_to_onnx`, but probably it's time to follow our own promises and at least not introduce a new deprecated stuff.
@shaydeci @Louis-Dupont WDYT?

Reply:
Why are you saing input_shape is deprecated?

Reply:
Oh I see now. 
Whatever you think...

Reply:
Fixed - Removed shape and using only "input_size" in CoreML